sklearn/model_selection/_search.py

```diff
@@ -552,17 +565,39 @@ def _fit(self, X, y, labels, parameter_iterable):
 
         # Computed the (weighted) mean and std for all the candidates
         weights = test_sample_counts if self.iid else None
-        means = np.average(test_scores, axis=1, weights=weights)
-        stds = np.sqrt(np.average((test_scores - means[:, np.newaxis]) ** 2,
-                                  axis=1, weights=weights))
+
+        time = np.array(_, dtype=np.float64).reshape(n_candidates, n_splits)
+        time_means = np.average(time, axis=1, weights=weights)
+        time_stds =  np.sqrt(
+                np.average((time - time_means[:, np.newaxis]) ** 2,
+                           axis=1, weights=weights))
+        if self.return_train_score:
+            train_means = np.average(train_scores, axis=1, weights=weights)
```

Comment:
Sorry if I'm a bit slow here, but please elucidate at some point. What's `iid` trying to hack? Why does accepting training data in the scorer help?